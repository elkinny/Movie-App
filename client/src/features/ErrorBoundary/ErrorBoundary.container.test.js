import ErrorBoundaryContainer from './ErrorBoundary.container';
import React from 'react';
import { shallow } from './../../test/enzymeSetup.js';

describe('ErrorBoundaryContainer: ', () => {
  it('1. getDerivedStateFromError() to return { hasError: true }', () => {
    expect(ErrorBoundaryContainer.getDerivedStateFromError()).toEqual({ hasError: true });
  });

  it('2. componentDidCatch() to call setState()', () => {
    const component = shallow(
      <ErrorBoundaryContainer errorName="errorName" componentStack="componentStack" />,
    );
    const inst = component.instance();
    const spy = jest.spyOn(inst, 'setState');
    inst.componentDidCatch('error', 'info');
    expect(spy).toHaveBeenCalled();
  });

  it('3. Render error message in component', () => {
    const component = shallow(
      <ErrorBoundaryContainer errorName="errorName" componentStack="componentStack" />,
    );
    component.setState({ hasError: true });
    expect(component).toMatchSnapshot();
  });

  it('4. Render children if no error', () => {
    const component = shallow(
      <ErrorBoundaryContainer errorName="errorName" componentStack="componentStack">
        Hi
      </ErrorBoundaryContainer>,
    );
    component.setState({ hasError: false });
    expect(component).toMatchSnapshot();
  });
});
