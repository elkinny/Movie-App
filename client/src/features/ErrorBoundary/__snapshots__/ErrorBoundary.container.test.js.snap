// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundaryContainer:  3. Render error message in component 1`] = `<ErrorBoundaryComponent />`;

exports[`ErrorBoundaryContainer:  4. Render children if no error 1`] = `"Hi"`;
